script/upgrade-st2-check

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env sh
+set -e
+
+DIR=$( dirname "$(readlink -f "$0")" )
+. $DIR/shared-functions
+
+TEST_PACKAGE="st2common"
+
+is_st2_installed() {
+    local __resultvar=$1
+    eval $__resultvar=false
+    if [ "$OS" = "Linux" ]; then
+        if [ "$DIST" = "Debian" ]; then
+            dpkg -s ${TEST_PACKAGE} >> /dev/null
+        elif [ "$DIST" = "RedHat" ]; then
+            rpm -qa ${TEST_PACKAGE} | grep ${TEST_PACKAGE} >> /dev/null
+        fi
+    fi
+
+    if [ $? -eq 0 ]; then
+        eval $__resultvar=true
+    fi
+}
+
+is_st2_installed WARN_ABOUT_UPDATE
+
+# If st2 is not installed then exit immediately
+if [ ${WARN_ABOUT_UPDATE} = false ]; then
+    exit 0
+fi
+
+PROJECT_ROOT=/opt/puppet
+ANSWERS_FILE_YAML=${PROJECT_ROOT}/hieradata/answers.yaml
+ANSWERS_FILE_JSON=${PROJECT_ROOT}/hieradata/answers.json
+ANSWERS_FILE=""
+
+if [ -f ${ANSWERS_FILE_YAML} ]; then
+    ANSWERS_FILE=${ANSWERS_FILE_YAML}
+elif [ -f ${ANSWERS_FILE_JSON} ]; then
+    ANSWERS_FILE=${ANSWERS_FILE_JSON}
+fi
+
+grep "st2::version" ${ANSWERS_FILE} >> /dev/null && grep "st2::revision" ${ANSWERS_FILE} >> /dev/null
+# Safe to exit if version and revision are defined in answers
+if [ $? -eq 0 ]; then
+    exit 0
+fi
+
+INSTALLED_VERSION_REVISION=""
+if [ "$OS" = "Linux" ]; then
+    if [ "$DIST" = "Debian" ]; then
+        INSTALLED_VERSION_REVISION=`dpkg -s ${TEST_PACKAGE} | grep Version | awk -F: '{print $2}' | sed -e 's/^[ \t]*//'`
+    elif [ "$DIST" = "RedHat" ]; then
+        INSTALLED_VERSION_REVISION=`rpm -qa ${TEST_PACKAGE} | sed -e "s/${TEST_PACKAGE}-//" | sed -e 's/.noarch//'`
+    fi
+fi
+
+HIERA_ST2_YAML=${PROJECT_ROOT}/hieradata/role/st2.yaml
+if [ ! -f ${HIERA_ST2_YAML} ]; then
+    echo "Could not find ${HIERA_ST2_YAML} so cannot perform version checks."
+    exit 0
```

Comment:
This would be a pretty strange condition to hit. Someone would have to willing delete the file to end up here but I do not know if installer should fail at this point though - it will likely blow up elsewhere.